### PR TITLE
Add $(HELPER_DIR) and others to Makefile.common

### PR DESCRIPTION
**Pull request by karkhaz**
This commit:

- Adds the $(HELPER_DIR) variable, set to the parent of the 'proofs'
  directory, to Makefile.common
- Renames the "sources" directory to "models", so that it doesn't get
  confused with $(SRCDIR)
- Adds the $(MODELS_DIR) and $(STUBS_DIR) variables for easy reference
  from proof makefiles
- Adds a recommendation that project owners should set $(INCLUDES) to
  contain "-I$(HELPER_DIR)/include". This commit does not actually set
  this variable, since the decision of which order it comes in in the
  include list is a matter of project policy.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: template-for-repository/proofs/Makefile-project-defines [hunk 1]
```diff
@@ -20,6 +20,11 @@
 # LINK_FLAGS =
 
 # Preprocessor include paths -I...
+# It is likely that you will want to add at least
+#     INCLUDES += -I$(HELPER_DIR)/include
+# but you will want to decide what order that comes in relative to the
+# other includes in your project.
+#
```
Comment (markrtuttle) [inline template-for-repository/proofs/Makefile-project-defines:27]: I'd shorten this to "Consider adding INCLUDES += ..."

File: template-for-repository/models/README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 CBMC proof source code
```
Comment (markrtuttle) [inline template-for-repository/models/README.md:1]: This renaming of sources to models is a bit gratuitous since the MQTT and HTTP folks have already started populating sources and includes with project code.  If it is likely people will think cbmc/sources is part of the source tree, it is equally likely that people will think cbmc/include is part of the source tree.  It might be worth using the names PROOF_SOURCE=cbmc/sources and PROOF_INCLUDE=cbmc/include in the Makefile.common.

File: template-for-repository/proofs/Makefile.common [hunk 3]
```diff
@@ -83,6 +83,9 @@ default: report
 # Absolute path to the directory containing this Makefile.common
 # See https://ftp.gnu.org/old-gnu/Manuals/make-3.80/html_node/make_17.html
 PROOF_ROOT = $(dir $(abspath $(filter %/Makefile.common,$(MAKEFILE_LIST))))
+HELPER_DIR = $(shell dirname $(PROOF_ROOT))
+STUBS_DIR = $(HELPER_DIR)/stubs
+MODELS_DIR = $(HELPER_DIR)/models
```
Comment (markrtuttle) [inline template-for-repository/proofs/Makefile.common:88]: Again, I'd use PROOF_STUB, PROOF_SOURCE, PROOF_INCLUDE.

HELPER_DIR is a name we commonly use, but it is pretty meaningless, and I intentionally did not use it.  I'd rather something like CBMC_ROOT for cbmc, and have PROOF_STUB, PROOF_SOURCE, PROOF_INCLUDE be the stubs, sources, include subdirs.

File: template-for-repository/proofs/Makefile-project-defines (hunk 1 above)
Comment (karkhaz) [reply template-for-repository/proofs/Makefile-project-defines:27]: Done

File: template-for-repository/proofs/Makefile.common (hunk 3 above)
Comment (karkhaz) [reply template-for-repository/proofs/Makefile.common]: I changed HELPER_DIR to CBMC_ROOT.
I changed STUBS_DIR to PROOF_STUB.
I changed MODELS_DIR to PROOF_SOURCE and renamed "models" back to "sources".

I did not add PROOF_INCLUDE because we don't know what the include order will be. The recommendation in the `Makefile-project-defines` file is now to add `-I$(CBMC_ROOT)/include` to the `INCLUDES` variable. I could set `PROOF_INCLUDE = $(CBMC_ROOT)/include` and recommend to add `PROOF_INCLUDE` to `INCLUDE`, but it seems unnecessary to create a variable that will only ever be used once (we will never append to `PROOF_INCLUDE` and never use it in any other context).

File: template-for-repository/models/README.md (hunk 2 above)
Comment (karkhaz) [reply template-for-repository/models/README.md]: Done
